src/esp/gfx/GenericDrawable.h

```diff
@@ -49,6 +50,8 @@ class GenericDrawable : public Drawable {
       shader_;
   Magnum::Resource<MaterialData, PhongMaterialData> materialData_;
   Magnum::Resource<LightSetup> lightSetup_;
+
+  Magnum::Shaders::Phong::Flags flags_;
```

Comment:
Before, the flag was computed again and again in each rendering loop, which I did not think it was necessary. So create a member variable to store it. It will be only computed once in the constructor.